src/js/tech/setup-sourceset.js

```diff
@@ -1,21 +1,121 @@
 import window from 'global/window';
 import mergeOptions from '../utils/merge-options';
 
-const setupSourceset = function(tech) {
+/**
+ * This function is used to fire a sourceset when there is something
+ * similar to `mediaEl.load()` being called. It will try to find the `src`
+ * and if there it cannot there was no `sourceset` and it will not fire one.
+ *
+ * @param {Html5} tech
+ *        The tech object that sourceset was setup on
+ *
+ * @return {boolean}
+ *         returns false if the sourceset was not fired and true otherwise.
+ */
+const sourcesetLoad = (tech) => {
+  const el = tech.el();
 
-  if (!tech.featuresSourceset) {
-    return;
+  // if `el.src` is set, that source will be loaded.
+  if (el.src) {
+    tech.triggerSourceset(el.src);
+    return true;
   }
 
-  const el = tech.el();
+  /**
+   * Since there isn't a src property on the media element, source elements will be used for
+   * implementing the source selection algorithm. This happens asynchronously and
+   * for most cases were there is more than one source we cannot tell what source will
+   * be loaded, without re-implementing the source selection algorithm. At this time we are not
+   * going to do that. There are three special cases that we do handle here though:
+   *
+   * 1. If there are no sources, do not fire `sourceset`.
```

Comment:
This changed the behavior on `load` slightly, but I think the comment does a good job explaining how.